### DISCUSSION TITLE
Alpine state not updating on Livewire Change

### DISCUSSION THREAD

**Discussion by tonz1818** [livewire/livewire#3694] [1. Help] [answered]
Hello, 

I was using livewire with Alpine 2 but recently updated to Alpine 3 and now my Alpine state wont change when livewire updates. 

`<a 
     x-data="{ url: '{{ route('tours.booking', ['session_id' => $firstSession]) }}' }"
     href="{{ route('tours.booking', ['session_id' => $firstSession]) }}"
     @selection-change.window="" 
     x-bind:href="url">
        Book
     </a>`

But when $firstSession updates via livewire, Alpine state url doesnt and remains the value on page load.

**Comment by joshhanley** [comment, marked as answer]
@tonz1818 Alpine v3 won't re-process data that is echoed on in `x-data`, so if you need to keep data in sync between Livewire and Alpine, I suggest using `@entangle` or firing off browser events from Livewire and listen for it in Alpine to update.

Hope this helps!

**Reply by tonz1818** [reply]
I was afraid that was going to be the solution. I really wanted to keep the server-client requested to a min. So I ended up going with native JS.
